### PR TITLE
Add support for Upsert operations to batching

### LINKED ISSUE #16250: Add Upsert support to Azure.Data.Tables TableTransactionalBatch
Package: `Azure.Data.Tables`

Can we add `Upsert` (merge/ replace) to the `TableTransactionalBatch` to match the `Add`, `Delete` and `Update` that currently exist.

### PR DESCRIPTION
**Pull request by christothes**
closes #16250

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableRestClient.cs [hunk 1]
```diff
@@ -166,7 +166,7 @@ public Response<List<Response>> SendBatchRequest(HttpMessage message, List<(ITab
 
         private static string AppendEntityInfoToMessage(string messsage, ITableEntity entity)
         {
-            return messsage += $"\n\nRowKey={entity.RowKey}";
+            return messsage += $"\nRowKey={entity.RowKey}\n";
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableRestClient.cs:169]: Do we typically have a new line at the end of the exception message?

File: sdk/tables/Azure.Data.Tables/tests/SessionRecords/TableClientLiveTests(CosmosTable)/BatchInsertAndMergeAndDelete.json [hunk 2]
```diff
@@ -130,72 +130,123 @@
...
...pEYXRhU2VydmljZVZlcnNpb246IDMuMA0KUHJlZmVyOiByZXR1cm4tbm8tY29udGVudA0KQWNjZXB0OiBhcHBsaWNhdGlvbi9qc29uDQpDb250ZW50LVR5cGU6IGFwcGxpY2F0aW9uL2pzb247b2RhdGE9bm9tZXRhZGF0YQ0KDQp7IlN0cmluZ1R5cGVQcm9wZXJ0eSI6IlRoaXMgaXMgdGFibGUgZW50aXR5IG51bWJlciAwNSIsIkRhdGV0aW1lVHlwZVByb3BlcnR5IjoiMjAyMC0wMS0wMVQwMTowNjowMC4wMDAwMDAwWiIsIkRhdGV0aW1lVHlwZVByb3BlcnR5QG9kYXRhLnR5cGUiOiJFZG0uRGF0ZVRpbWUiLCJEYXRldGltZU9mZnNldFR5cGVQcm9wZXJ0eSI6IjIwMjAtMDEtMDFUMDE6MDY6MDAuMDAwMDAwMFoiLCJEYXRldGltZU9mZnNldFR5cGVQcm9wZXJ0eUBvZGF0YS50eXBlIjoiRWRtLkRhdGVUaW1lIiwiR3VpZFR5cGVQcm9wZXJ0eSI6IjBkMzkxZDE2LTk3ZjEtNGI5YS1iZTY4LTRjYzg3MWY5MDAwNSIsIkd1aWRUeXBlUHJvcGVydHlAb2RhdGEudHlwZSI6IkVkbS5HdWlkIiwiQmluYXJ5VHlwZVByb3BlcnR5IjoiQVFJREJBVT0iLCJCaW5hcnlUeXBlUHJvcGVydHlAb2RhdGEudHlwZSI6IkVkbS5CaW5hcnkiLCJJbnQ2NFR5cGVQcm9wZXJ0eSI6IjUiLCJJbnQ2NFR5cGVQcm9wZXJ0eUBvZGF0YS50eXBlIjoiRWRtLkludDY0IiwiRG91YmxlVHlwZVByb3BlcnR5Ijo1LCJEb3VibGVUeXBlUHJvcGVydHlAb2RhdGEudHlwZSI6IkVkbS5Eb3VibGUiLCJJbnRUeXBlUHJvcGVydHkiOjUsIlBhcnRpdGlvbktleSI6InNvbVBhcnRpdGlvbiIsIlJvd0tleSI6IjA1IiwiVGltZXN0YW1wIjpudWxsfQ0KLS1jaGFuZ2VzZXRfNTU1OTRmMjQtMDQ4YS05N2NmLWJjODAtZTM1YjcyMjBmMTU5LS0NCg0KLS1iYXRjaF80MTU0YmY0NS1lYTRkLTY0YTUtOTNmNC00ZDM2ODdjYzk3ODMtLQ0K",
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/tests/SessionRecords/TableClientLiveTests(CosmosTable)/BatchInsertAndMergeAndDelete.json:210]: It's unfortunate that we end-up with multipart as binary but I guess it's to the spec.

**Review by pakrym — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableRestClient.cs (hunk 1 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableRestClient.cs:169]: Not sure - I added it while testing when I saw how it blended with the next line when the test initially failed. Do you think it should be removed?

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableRestClient.cs]: removed it and replaced with a trailing `.`

**Pull request merged**